project/apps/user-service/src/adapters/db/users.supabase.ts

```diff
@@ -84,14 +84,15 @@ export class SupabaseUsersRepository implements UsersRepository {
     } as UserCollectionDto;
   }
 
-  async findById(id: string): Promise<UserDataDto> {
+  async findById(id: string): Promise<UserDataDto | null> {
     const { data, error } = await this.supabase
       .from(this.PROFILES_TABLE)
       .select()
       .eq('id', id)
       .single<UserDataDto>();
 
-    if (error) {
+    // for any unexpected error apart from null
+    if (error && error.code != 'PGRST116') {
       throw error;
     }
```

Comment:
```suggestion
    if (error) {
      throw error;
    }
```
continuation from above